### PR TITLE
Add search layout

### PR DESCRIPTION
**Pull request by OleksiiKislukhin**

### CODE REVIEW

**Review by iyehorova — APPROVED** [review]
👀

**Comment by AndriiZakharenko** [comment]
👀

**Review by AndriiZakharenko — APPROVED** [review]

File: src/modules/ProductsPage/ProductsPage.tsx [hunk 1]
```diff
@@ -29,9 +30,17 @@ export const ProductsPage = () => {
   const isAggregatesLoading = useAppSelector(selectAggregateLoading);
   const isLoading = isProductsLoading || isAggregatesLoading;
 
-  const productList = Object.values(products[productsCategory]);
+  const [productList, setProductList] = useState(
```
Comment (iyehorova) [inline src/modules/ProductsPage/ProductsPage.tsx]: We are using redux for the product and it doesn't need to use state

File: src/modules/ProductsPage/ProductsPage.tsx [hunk 2]
```diff
@@ -44,6 +53,22 @@ export const ProductsPage = () => {
...
+  useEffect(() => {
+    const newProducts = productList.filter(product => {
+      const searchBy = searchByName
+        ? searchByName.trim().toLocaleLowerCase()
+        : '';
+
+      return product.name.toLocaleLowerCase().includes(searchBy);
+    });
+
+    setSearchFilteredProducts(newProducts);
```
Comment (iyehorova) [inline src/modules/ProductsPage/ProductsPage.tsx]: Apply useFilteredProducts instead this useEffect

**Review by iyehorova — CHANGES_REQUESTED** [review]
Please, check comment below

**Review by iyehorova — APPROVED** [review]
👀

**Pull request merged**